### DISCUSSION TITLE
Is using UUIDs for session-tokens secure enough?

### DISCUSSION THREAD

**Discussion by tirumaraiselvan** [nextauthjs/next-auth#10620] [Help]
After authentication, the session token is stored as `next-auth.session-token` key in the cookie but it's value is simply a UUID. This doesn't guarantee good entropy and is not considered a good (let alone best) practise: 

https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html

https://stackoverflow.com/questions/5244455/best-practices-for-sessionid-authentication-token-generation

Should there be ways to configure the session ID generation or is there anything in the roadmap to improve upon this?

**Comment by mid2ly** [comment]
I think it's safe enough, but if you think it's not, you can try the following:

```typescript
const defaultAdapter = PrismaAdapter(prisma) // Change to the adapter you want to use.

// Parameter entered in the `NextAuth(config)` section
export const config = {
    adapter: {
        ...defaultAdapter,
        async createSession(session: { sessionToken: string; userId: string; expires: Date }) {
            // Passes the overridden session to the default adapter.
            return defaultAdapter.createSession!({
                ...session,
                // Overrides the session token.
                sessionToken: crypto.randomBytes(48).toString('hex') // or write secure random token logic.
            })
        }
    }
} satisfies NextAuthConfig
```
